protobuf_release.bzl

```diff
@@ -0,0 +1,34 @@
+"""
+Generates package naming variables for use with rules_pkg.
+"""
+
+load("@rules_pkg//:providers.bzl", "PackageVariablesInfo")
+load("@bazel_tools//tools/cpp:toolchain_utils.bzl", "find_cpp_toolchain")
+load(":protobuf_version.bzl", "PROTOBUF_VERSION")
+
+def _package_naming_impl(ctx):
+  values = {}
+  values["product_name"] = ctx.attr.product_name
+  values["version"] = PROTOBUF_VERSION
+
+  # infer from the current cpp toolchain.
+  toolchain = find_cpp_toolchain(ctx)
+  values["cpu"] = toolchain.cpu
+  
+  return PackageVariablesInfo(values = values)
+
+
+package_naming = rule(
+  implementation = _package_naming_impl,
+    attrs = {
+      "product_name": attr.string(
+          doc = "Placeholder for our final product name.",
+      ),
+      "version": attr.string(
+          doc = "Placeholder for our release version.",
+      ),
+      "_cc_toolchain": attr.label(default = Label("@bazel_tools//tools/cpp:current_cc_toolchain")),
```

Comment:
This variable is unused, what is it for?

Reply:
This is necessary to get an instance of the toolchain that is currently in use, the find call I'm using in the impl relies on this existing.

Reply:
Can we add that context as a comment?